### PR TITLE
[Storage] [DataMovement] Made the respective members `public` to `protected internal` members

### PR DESCRIPTION
**Pull request by amnguye**
Made the following members `public` to `protected internal` members (including all derived classes):
    - `StorageResource.CanProduceUri`
    - `StorageResource.IsContainer`
    - `StorageResourceContainer.GetStorageResourcesAsync`
    - `StorageResourceSingle.Length`
    - `StorageResourceSingle.MaxChunkSize`
    - `StorageResourceSingle.ResourceId`
    - `StorageResourceSingle.TransferType`
    - `StorageResourceSingle.CompleteTransferAsync`
    - `StorageResourceSingle.CopyBlockFromUriAsync`
    - `StorageResourceSingle.CopyFromUriAsync`
    - `StorageResourceSingle.DeleteIfExistsAsync`
    - `StorageResourceSingle.GetCopyAuthorizationHeaderAsync`
    - `StorageResourceSingle.GetPropertiesAsync`
    - `StorageResourceSingle.ReadStreamAsync`
    - `StorageResourceSingle.WriteFromStreamAsync`

Made the following members `public` to `protected` members (including all derived classes):
    - `BlobStorageResourceContainer.CanProduceUri`
    - `BlobStorageResourceContainer.GetStorageResourcesAsync`
    - `*BlobStorageResource.CanProduceUri`
    - `*BlobStorageResource.Length`
    - `*BlobStorageResource.MaxChunkSize`
    - `*BlobStorageResource.ResourceId`
    - `*BlobStorageResource.TransferType`
    - `*BlobStorageResource.CompleteTransferAsync`
    - `*BlobStorageResource.CopyBlockFromUriAsync`
    - `*BlobStorageResource.CopyFromUriAsync`
    - `*BlobStorageResource.DeleteIfExistsAsync`
    - `*BlobStorageResource.GetCopyAuthorizationHeaderAsync`
    - `*BlobStorageResource.GetPropertiesAsync`
    - `*BlobStorageResource.ReadStreamAsync`
    - `*BlobStorageResource.WriteFromStreamAsync`

I also added Mock*BlobStorageResource because we could not access the `protected` members from our unit tests. If someone has a better solution to doing this, I'm open to suggestions.

### CODE REVIEW

File: sdk/storage/Azure.Storage.DataMovement/src/StorageResource.cs [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 using System;
 
+#pragma warning disable SA1402  // File may only contain a single type
```
Comment (jalauzon-msft) [inline sdk/storage/Azure.Storage.DataMovement/src/StorageResource.cs:6]: Is this needed?

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs [hunk 2]
```diff
@@ -7,6 +7,7 @@
 using Azure.Storage.Blobs.Models;
 using Azure.Storage.DataMovement.Models;
 using Azure.Storage.DataMovement.Models.JobPlan;
+using System;
```
Comment (jalauzon-msft) [inline sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs:10]: nit: Can probably revert this?

**Review by jalauzon-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.DataMovement/src/StorageResource.cs (hunk 1 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement/src/StorageResource.cs]: removed

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs (hunk 2 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs]: yup done

**Pull request merged**